src/app/clusters/resource-monitoring-server/resource-monitoring-server.cpp

```diff
@@ -0,0 +1,300 @@
+{
+    if (aNewChangeIndication == chip::app::Clusters::ResourceMonitoring::ChangeIndicationEnum::kWarning)
+    {
+        if (!HasFeature(ResourceMonitoring::Feature::kWarning))
+        {
+            return Protocols::InteractionModel::Status::InvalidValue;
+        }
+    }
+    auto oldChangeIndication = mChangeIndication;
+    mChangeIndication        = aNewChangeIndication;
+    if (mChangeIndication != oldChangeIndication)
+    {
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::ChangeIndication::Id);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+chip::Protocols::InteractionModel::Status Instance::UpdateInPlaceIndicator(bool aNewInPlaceIndicator)
+{
+    auto oldInPlaceIndicator = mInPlaceIndicator;
+    mInPlaceIndicator        = aNewInPlaceIndicator;
+    if (mInPlaceIndicator != oldInPlaceIndicator)
+    {
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::InPlaceIndicator::Id);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+chip::Protocols::InteractionModel::Status Instance::UpdateLastChangedTime(DataModel::Nullable<uint32_t> aNewLastChangedTime)
+{
+    auto oldLastchangedTime = mLastChangedTime;
+    mLastChangedTime        = aNewLastChangedTime;
+    if (mLastChangedTime != oldLastchangedTime)
+    {
+        chip::app::GetAttributePersistenceProvider()->WriteScalarValue(
+            ConcreteAttributePath(mEndpointId, mClusterId, Attributes::LastChangedTime::Id), mLastChangedTime);
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::LastChangedTime::Id);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+uint8_t Instance::GetCondition() const
+{
+    return mCondition;
+}
+chip::app::Clusters::ResourceMonitoring::ChangeIndicationEnum Instance::GetChangeIndication() const
+{
+    return mChangeIndication;
+}
+
+chip::app::Clusters::ResourceMonitoring::DegradationDirectionEnum Instance::GetDegradationDirection() const
+{
+    return mDegradationDirection;
+}
+
+bool Instance::GetInPlaceIndicator() const
+{
+    return mInPlaceIndicator;
+}
+
+DataModel::Nullable<uint32_t> Instance::GetLastChangedTime() const
+{
+    return mLastChangedTime;
+}
+
+bool Instance::IsValidAliasCluster() const
+{
+    for (unsigned int AliasedCluster : AliasedClusters)
+    {
+        if (mClusterId == AliasedCluster)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+void Instance::HandleResetCondition(HandlerContext & ctx,
+                                    const ResourceMonitoring::Commands::ResetCondition::DecodableType & commandData)
+{
+
+    Status resetConditionSuccessful = OnResetCondition();
+    if (Status::Success != resetConditionSuccessful)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, resetConditionSuccessful);
+        return;
+    }
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Success);
```

Comment:
Is the only reason this is not all a single `ctx.mCommandHandler.AddStatus(ctx.mRequestPath, resetConditionSuccessful);` because of the logging?  But then it might be clearer to just condition the logging on the status being success, then have the one AddStatus call.

Reply:
That's a left over when this method was I bit more complicated. You're right, this doesn't
make that much sense the way it is. 
Simplified it.